observability/src/main/java/io/grpc/protobuf/observability/LogRecordExtension.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2022 The gRPC Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.grpc.protobuf.observability;
+
+import io.grpc.observabilitylog.v1.GrpcLogRecord;
+import java.util.logging.Level;
+import java.util.logging.LogRecord;
+
+
+public class LogRecordExtension extends LogRecord {
+  private GrpcLogRecord grpcLogRecord;
```

Comment:
One more thing that needs to be thought through (may be not for this PR but for future): how are callers (users of this class) going to set individual fields of `GrpcLogRecord` ? Are they going to use the getter `getGrpcLogRecord` and simply set the fields of `grpcLogRecord`? Or should we provider individual setters? For `getGrpcLogRecord` should we provide a read-only copy? Things to think about

Reply:
For now,  user sets the fields individually by using `getGrpcLogRecordBuilder()`  and calls `setGrpcLogRecord(record)` to add record to `LogRecordExtension`.